app/profile/components/ProfileForm.tsx

```
@@ -0,0 +1,364 @@
+    error: null as string | null,
+    success: null as string | null,
+  });
+
+  // Separate state for each child form
+  const [childrenStates, setChildrenStates] = useState<Record<string, {
+    isLoading: boolean;
+    error: string | null;
+    success: string | null;
+  }>>(
+    userChildren.reduce((acc, uc) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)